packages/identity-service/src/utils/fpHelpers.js

```diff
@@ -22,6 +24,72 @@ const getDeviceIDCountForUserId = async (userId) => {
   return res.length
 }
 
+const validateFingerprint = async (email, visitorId, req) => {
+  try {
+    if (!visitorId) {
+      req.logger.info(
+        `Fingerprint not validated for ${email}: no visitorId provided`
+      )
+      return
+    }
+
+    const { blockchainUserId: userId } = await models.User.findOne({
+      where: { email }
+    })
+    if (!userId) {
+      req.logger.info(
+        `Fingerprint not validated for ${email}: no matching userId found`
+      )
+      return
+    }
+
+    const {
+      visits: [latestVisit]
+    } = await req.app.get('fpClient').getVisitorHistory(visitorId)
+
+    const tenMinutesInMilliseconds = 10 * 60 * 1000
+    const now = Date.now()
+    if (now - new Date(latestVisit.timestamp) > tenMinutesInMilliseconds) {
```

Comment:
can latestVisit potentially be undefined and this throws? maybe that's desired behavior